### ISSUE TITLE
core: ap-southeast-4 is missing in AWS_REGIONS_AND_RULES

### ISSUE THREAD

**Issue by zxkane** [aws/aws-cdk#25796]
### Describe the bug

Create a simple CDK app with Step Functions like below

```ts
import { App, Stack, StackProps, Duration } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { Wait, WaitTime, Fail, StateMachine } from 'aws-cdk-lib/aws-stepfunctions';

export class MyStack extends Stack {
  constructor(scope: Construct, id: string, props: StackProps = {}) {
    super(scope, id, props);

    const waitX = new Wait(this, 'Wait X Seconds', {
      time: WaitTime.secondsPath('$.waitSeconds'),
    });
    
    
    
    const jobFailed = new Fail(this, 'Job Failed', {
      cause: 'AWS Batch Job Failed',
      error: 'DescribeJob returned FAILED',
    });
    
    
    const definition = waitX
      .next(jobFailed);
    
    new StateMachine(this, 'StateMachine', {
      definition,
      timeout: Duration.minutes(5),
    });
  }
}

const app = new App();

new MyStack(app, 'simple-sfn', {  });

app.synth();
```

Deploy it then got below error,

```
 ❌ Deployment failed: Error [ValidationError]: Template error: Unable to get mapping for ServiceprincipalMap::ap-southeast-4::states
    at Request.extractError (/home/ubuntu/environment/tmp/s3-default-acl/node_modules/aws-cdk/lib/index.js:292:46245)
    at Request.callListeners (/home/ubuntu/environment/tmp/s3-default-acl/node_modules/aws-cdk/lib/index.js:292:89237)
    at Request.emit (/home/ubuntu/environment/tmp/s3-default-acl/node_modules/aws-cdk/lib/index.js:292:88685)
    at Request.emit (/home/ubuntu/environment/tmp/s3-default-acl/node_modules/aws-cdk/lib/index.js:292:195114)
    at Request.transition (/home/ubuntu/environment/tmp/s3-default-acl/node_modules/aws-cdk/lib/index.js:292:188666)
    at AcceptorStateMachine.runTo (/home/ubuntu/environment/tmp/s3-default-acl/node_modules/aws-cdk/lib/index.js:292:153538)
    at /home/ubuntu/environment/tmp/s3-default-acl/node_modules/aws-cdk/lib/index.js:292:153868
    at Request.<anonymous> (/home/ubuntu/environment/tmp/s3-default-acl/node_modules/aws-cdk/lib/index.js:292:188958)
    at Request.<anonymous> (/home/ubuntu/environment/tmp/s3-default-acl/node_modules/aws-cdk/lib/index.js:292:195189)
    at Request.callListeners (/home/ubuntu/environment/tmp/s3-default-acl/node_modules/aws-cdk/lib/index.js:292:89405) {
  code: 'ValidationError',
  time: 2023-05-31T15:34:51.679Z,
  requestId: 'ffe2493d-e15b-48b2-9870-0988aa7ac7f2',
  statusCode: 400,
  retryable: false,
  retryDelay: 89.89593283858954
```

### Expected Behavior

The app can be deployed without errors.

### Current Behavior

see above

### Reproduction Steps

see above

### CDK CLI Version

2.81.0

### Node.js Version

16.20.0

### OS

ubuntu 18.04

### Language

Typescript

**Comment by pahud** [comment]
@zxkane 

I can't reproduce it. Can you paste the full CDK synth output in the comment?

**Comment by zxkane** [comment]
@pahud  I found there is a feature flag `@aws-cdk/aws-iam:standardizedServicePrincipals` could remove above `ServiceprincipalMap`, it's also safe for my use case.

https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/cx-api/FEATURE_FLAGS.md#aws-cdkaws-iamstandardizedserviceprincipals

However, my application specifies [the log rention period of Lambda function](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_lambda.Function.html#logretention), it introduceds another map named `DefaultCrNodeVersionMap` missing the `ap-southeast-4` too.

https://github.com/aws/aws-cdk/blob/7f3152a917f716ee64b496b42f1dad76bc33e188/packages/aws-cdk-lib/aws-logs/lib/log-retention.ts#L174

**Comment by pahud** [comment]
@zxkane 

Yeah. I guess we just missed `ap-southeast-4` [here](https://github.com/aws/aws-cdk/blob/60699f4a712755451c144e4af6782c505f3a0b63/packages/aws-cdk-lib/region-info/lib/aws-entities.ts#L27-L63).

**Comment by pahud** [comment]
to be closed by https://github.com/aws/aws-cdk/issues/25890

**Comment by colifran** [comment]
Closed by [#26919](https://github.com/aws/aws-cdk/pull/26919)
